orttraining/orttraining/training_ops/cuda/optimizer/lamb.cu

```diff
@@ -10,6 +16,18 @@
 
 namespace onnxruntime {
 namespace cuda {
+
+__forceinline__ __host__ __device__ int least_pow2_bound(int value) {
+  unsigned int value_ = static_cast<unsigned int>(value);
+  --value_;
+  value_ |= value_ >> 1;
+  value_ |= value_ >> 2;
+  value_ |= value_ >> 4;
+  value_ |= value_ >> 8;
+  value_ |= value_ >> 16;
+  return static_cast<int>(++value_);
+}
```

Comment:
maybe making use of some CUDA integer intrinsics?
[https://docs.nvidia.com/cuda/cuda-math-api/group__CUDA__MATH__INTRINSIC__INT.html#group__CUDA__MATH__INTRINSIC__INT](https://docs.nvidia.com/cuda/cuda-math-api/group__CUDA__MATH__INTRINSIC__INT.html#group__CUDA__MATH__INTRINSIC__INT)

Reply:
It has been defined in `/bert_ort/wechi/onnxruntime1/onnxruntime/core/providers/cuda/reduction/reduction_utils.cuh`

Reply:
Removed this code snippet and added the appropriate header.

Reply:
Don't have to be in this PR, but maybe a chance to improve in reduction_utils.cuh

Reply:
Maybe this: ```return 1 << (32 - __clz(value_ - 1));```